### PR TITLE
Updated mode picker to use collection view

### PR DESCRIPTION
**Pull request by pnuts2**
Closes RM15993

### CODE REVIEW

File: Sources/TripKitUI/views/TKUIModePicker.swift [hunk 1]
```diff
@@ -56,10 +43,62 @@ public class TKUIModePicker<Item>: UIView where Item: TKUIModePickerItem {
...
+    prepareCollectionView()
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  private func prepareCollectionView() {
+    let collectionView = UICollectionView(frame: .zero, collectionViewLayout: UICollectionViewFlowLayout())
+    collectionView.showsHorizontalScrollIndicator = false
```
Comment (nighthawk) [inline Sources/TripKitUI/views/TKUIModePicker.swift:67]: I vaguely recall that there's a way to disable scrolling completely on a collection view (or scroll view?) and that wouldn't be necessary then.

File: Sources/TripKitUI/views/TKUIModePickerCell.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    super.awakeFromNib()
+    
+    contentWrapper.backgroundColor = .clear
+    contentWrapper.layer.cornerRadius = contentWrapper.frame.height * 0.5
+    
+    leftImageView.layer.cornerCurve = .continuous
+    leftImageView.layer.cornerRadius = leftImageView.frame.width * 0.5
+    
+    rightImageView.layer.cornerCurve = .continuous
+    rightImageView.layer.cornerRadius = rightImageView.frame.width * 0.5
```
Comment (nighthawk) [inline Sources/TripKitUI/views/TKUIModePickerCell.swift:40]: Can this cell resize? If so, you might need to do the same as also on `didSet` of the frame.

**Review by nighthawk — APPROVED** [review]
Looks good to me, just some minor comments.

**Comment by nighthawk** [comment]
@pnuts2 Tried this in TripGo in Sydney, but something is broken:

PNG image

**Comment by pnuts2** [comment]
> @pnuts2 Tried this in TripGo in Sydney, but something is broken:
> 
> PNG image

Yes, I've seen that previously, and thought I have fixed it. Will take a look at this again once I wrapped up the Feonix work on converting from button press to cell tap

**Review by nighthawk — CHANGES_REQUESTED** [review]
See bug/screenshot from the conversation. Brian is on it.

**Review by nighthawk — APPROVED** [review]
Works after all after I updated TripGo. Sorry for the confusion.

**Pull request merged**